### PR TITLE
TF-654 Show the progress of download attachment in app

### PR DESCRIPTION
**Pull request by dab246**
#654 

### Dependency 

- Upgrade version `dio` for `jmap-dart-client` at [PR#33](https://github.com/linagora/jmap-dart-client/pull/33)

### Resolved

https://user-images.githubusercontent.com/80730648/183249742-17d4e1fc-ed2c-4744-9316-428e7af32440.mov

### CODE REVIEW

File: core/lib/data/network/download/download_manager.dart [hunk 1]
```diff
@@ -82,39 +81,28 @@ class DownloadManager {
...
-      }
+      html.document.body?.children.remove(anchor);
+      html.Url.revokeObjectUrl(url);
 
+      return true;
+    } catch (error) {
+      log('DownloadManager::createAnchorElementDownloadFileWeb(): ERROR: $error');
       return false;
-    } catch (exception) {
-      throw exception;
```
Comment (hoangdat) [inline core/lib/data/network/download/download_manager.dart:117]: why you not throw exception?

**Review by ManhNTX — APPROVED** [review]

File: core/lib/data/network/download/download_manager.dart (hunk 1 above)
Comment (dab246) [reply core/lib/data/network/download/download_manager.dart:117]: I'm not handling any error events here at the moment. But maybe I'll add it so I can use it later.

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 2]
```diff
@@ -578,6 +598,7 @@ class MailboxDashBoardController extends ReloadableController {
     _connectivityStreamSubscription.cancel();
     _progressStateController.close();
     _isolateExecutor.dispose();
+    Get.delete<DownloadController>();
```
Comment (hoangdat) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:601]: why we need to do that? `Get` does not support us?

Comment (dab246) [reply lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:601]: This controller is not attached to the view so `Getx` will not automatically clean it.

**Review by hoangdat — APPROVED** [review]

**Pull request merged**